sky/packages/sky/lib/widgets/mimic.dart

```diff
@@ -0,0 +1,129 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:sky/widgets/basic.dart';
+
+class Mimic extends StatefulComponent {
+  Mimic({ Key key, this.original }) : super(key: key);
+
+  GlobalKey original;
+
+  void initState() {
+    _requestToStartMimic();
+  }
+
+  void syncFields(Mimic source) {
+    if (original != source.original) {
+      _stopMimic();
+      original = source.original;
+      _requestToStartMimic();
+    }
+  }
+
+  void didMount() {
+    super.didMount();
+    _requestToStartMimic();
+  }
+
+  void didUnmount() {
+    super.didUnmount();
+    _stopMimic();
+  }
+
+  Mimicable _mimicable;
+  Size _size;
+
+  void _requestToStartMimic() {
+    if (_mimicable != null)
```

Comment:
I think that this implies that we're already mimicking a widget. So this is likely an error; a comment and an assert would be helpful.